tesseract_core/runtime/mpa.py

```diff
@@ -0,0 +1,210 @@
+# Copyright 2025 Pasteur Labs. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+import csv
+import json
+import os
+import shutil
+from collections.abc import Generator
+from contextlib import contextmanager
+from contextvars import ContextVar
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Optional
+
+import requests
+
+
+class BaseBackend:
+    """Base class for MPA backends."""
+
+    def log_parameter(self, key: str, value: Any) -> None:
+        """Log a parameter."""
+        raise NotImplementedError
+
+    def log_metric(self, key: str, value: float, step: Optional[int] = None) -> None:
+        """Log a metric."""
+        raise NotImplementedError
+
+    def log_artifact(self, local_path: str) -> None:
+        """Log an artifact."""
+        raise NotImplementedError
+
+    def start_run(self) -> None:
+        """Start a new run."""
+        pass
+
+    def end_run(self) -> None:
+        """End the current run."""
+        pass
+
+
+class FileBackend(BaseBackend):
```

Comment:
does the FileBackend not implement the `start/end_run` API? "Not yet" is a fine answer as far as this PR is concerned, I think.